common/check_internal.h

```diff
@@ -43,6 +43,35 @@ CheckCondition(bool condition)
                                 const char* condition_str,
                                 llvm::StringRef extra_message) -> void;
 
+// A backport of std::is_scoped_enum_v from C++23.
+template <typename T, bool IsEnum = std::is_enum_v<T>>
+static constexpr bool IsScopedEnum = false;
+
+template <typename T>
+static constexpr bool IsScopedEnum<T, true> =
+    !std::convertible_to<T, std::underlying_type_t<T>>;
+
+// Allow converting format values; the default behaviour is to just pass them
+// through.
+template <typename T>
+auto ConvertFormatValue(T&& t) -> T&& {
+  return std::forward<T>(t);
+}
+
+// Convert enums to larger integers so that byte-sized enums are not confused
+// with being chars and printed as invalid (or nul-terminating) characters.
```

Comment:
Comment that you are excluding scoped enums because those have their own printing logic?

Reply:
Heh, clang-tidy does not understand partial specialization of constexpr variables and complains about the IsScopedEnum. Thinking over night, I realized we could just include scoped enums here with an explicit cast, and then the user doesn't have to write a cast (and possibly end up with a char by mistake). So I've done that and added a comment.